core/src/main/java/org/elasticsearch/search/suggest/SuggestionBuilder.java

```diff
@@ -212,6 +223,8 @@ protected void populateCommonFields(MapperService mapperService,
                 throw new IllegalArgumentException("Analyzer [" + luceneAnalyzer + "] doesn't exists");
             }
             suggestionContext.setAnalyzer(luceneAnalyzer);
+        } else {
+            suggestionContext.setAnalyzer(mapperService.searchAnalyzer());
         }
```

Comment:
The precedence should be given_analyzer -> default field analyzer -> default global analyzer for both `term` & `phrase` suggester. That means the field type has to be resolved before setting the default analyzer, if none was provided. 

Maybe we can add the following for resolving analyzer in `populateCommonFields` and remove the check from https://github.com/elastic/elasticsearch/blob/feature-suggest-refactoring/core/src/main/java/org/elasticsearch/search/suggest/phrase/PhraseSuggestParser.java#L166?
```java
MappedFieldType fieldType = mapperService.fullName(fieldname);
if (fieldType == null) {
    throw new IllegalArgumentException("no mapping found for field [" + fieldname + "]");
} else if (analyzer == null) {
    // no analyzer name passed in, so try the field's analyzer, or the default analyzer
    if (fieldType.searchAnalyzer() == null) {
        suggestionContext.setAnalyzer(mapperService.searchAnalyzer());
    } else {
        suggestionContext.setAnalyzer(fieldType.searchAnalyzer());
    }
} else {
    Analyzer luceneAnalyzer = mapperService.analysisService().analyzer(analyzer);
    if (luceneAnalyzer == null) {
        throw new IllegalArgumentException("analyzer [" + analyzer + "] doesn't exists");
    }
    suggestionContext.setAnalyzer(luceneAnalyzer);
}
```
Thinking out loud, maybe we should just have a set/get `MappedFieldType fieldType` in `SuggestionContext` instead of set/get `String fieldName` in the future?  

`completion` suggester only allows the index and search analyzers to be defined at mapping time (https://github.com/elastic/elasticsearch/blob/feature-suggest-refactoring/core/src/main/java/org/elasticsearch/index/mapper/core/CompletionFieldMapper.java#L164), and disregards the `analyzer` in `SuggestionContext` completely.